### PR TITLE
Update bzlmod example to use latest Bazel release 

### PR DESCRIPTION
**Pull request by kormide**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

The existing bzlmod example does not support the latest version of Bazel.

## What is the new behavior?

Bzlmod example uses a prerelease version of Bazel 6 and properly overrides the module path during integration testing.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by kormide** [comment]
FYI @alexeagle @Wyverald

File: tools/bazel_integration_test/test_runner.py [hunk 1]
```diff
@@ -52,6 +52,19 @@ def main(conf_file):
                     "--override_repository=rules_python=%s/rules_python"
                     % os.environ["TEST_SRCDIR"]
                 )
+                bazel_args.append(
+                    "--override_module=rules_python=%s/rules_python"
+                    % os.environ["TEST_SRCDIR"]
+                )
```
Comment (kormide) [inline tools/bazel_integration_test/test_runner.py:58]: This is the equivalent of `--override_repository` but for MODULE deps.

File: tools/bazel_integration_test/test_runner.py [hunk 2]
```diff
@@ -52,6 +52,19 @@ def main(conf_file):
...
+
+                # TODO: Despite not declaring dependencies on these in MODUILE.bzl
+                # or in the bzlmod example's MODULE.bzl, bazel is complaining about
+                # the following versions being yanked. For now, allow all yanked
+                # versions in integration tests.
+                #
+                #   protobuf@3.19.2
+                #   zlib@1.2.12
+                bazel_args.append("--allow_yanked_versions=all")
```
Comment (kormide) [inline tools/bazel_integration_test/test_runner.py:67]: @Wyverald Any idea why this was required? Those versions are indeed yanked in the BCR, but I don't have any direct or transitive dependencies on them via bzlmod (that I know of).

File: examples/BUILD [hunk 3]
```diff
@@ -40,4 +40,5 @@ bazel_integration_test(
 
 bazel_integration_test(
     name = "bzlmod_example",
+    override_bazel_version = "6.0.0-pre.20221012.2",
```
Comment (alexeagle) [inline examples/BUILD:43]: there's a 6.0.0-rc0 this morning, maybe use that?

**Review by alexeagle — APPROVED** [review]
Won't merge until we get an answer on the yanked problem

File: examples/BUILD (hunk 3 above)
Comment (kormide) [reply examples/BUILD:43]: That release tag doesn't appear to be available on GitHub. I think the one I'm referencing here is the latest.

File: tools/bazel_integration_test/test_runner.py (hunk 2 above)
Comment (fmeum) [reply tools/bazel_integration_test/test_runner.py]: It works with `last_green` and was fixed by https://github.com/bazelbuild/bazel/commit/d6d6183dc17484a8d5addcd68a7480ce7a5a97f6. While your module didn't depend on the yanked `zlib`, `@bazel_tools` did.

File: examples/BUILD (hunk 3 above)
Comment (Wyverald) [reply examples/BUILD]: it should be 6.0.0-rc1

Comment (Wyverald) [reply examples/BUILD]: sorry, "6.0.0rc1" without the dash

File: tools/bazel_integration_test/test_runner.py (hunk 2 above)
Comment (Wyverald) [reply tools/bazel_integration_test/test_runner.py]: Yep -- `bazel_tools` is a built-in module that depends on stuff like `rules_cc`, `rules_java`, etc. Unfortunately it also depends on these two yanked versions, but using 6.0.0rc1 should fix this.

Comment (kormide) [reply tools/bazel_integration_test/test_runner.py]: That works, thanks!

File: examples/BUILD (hunk 3 above)
Comment (alexeagle) [reply examples/BUILD]: oops sorry I was too fast and inaccurate :)

**Review by alexeagle — APPROVED** [review]

**Pull request merged**